tutorials/quick_start/keras_fw/keras_applications/model_lib_keras_applications.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from typing import Dict
+import tensorflow as tf
+
+from tutorials.quick_start.common.model_lib import BaseModelLib
+from tutorials.quick_start.keras_fw.utils import classification_eval, get_representative_dataset, separate_preprocess_model
+from tutorials.quick_start.common.constants import MODEL_NAME, BATCH_SIZE, VALIDATION_SET_LIMIT, VALIDATION_DATASET_FOLDER, IMAGENET_DATASET
+
+from tutorials.quick_start.common.results import DatasetInfo
+
+
+class ModelLib(BaseModelLib):
+    """
+    A class for quantizing pre-trained models from the tensorflow.keras.applications library
+    """
+    MISPLACED_CLASSES = ['MobileNetV3Small', 'MobileNetV3Large']
+
+    def __init__(self, args: Dict):
+        """
+        Init the ModelLib with user arguments
+        Args:
+            args (dict): user arguments
+        """
+        model_fn, self.model_package = self.get_keras_apps_model(args[MODEL_NAME])
+        # create model
+        self.model = model_fn()
+        self.preprocess = self.generate_preprocess()
+        self.dataset_name = IMAGENET_DATASET
+        # Extract image input size and verify it's not dynamic (equals None)
+        self.input_size = self.model.layers[0].input_shape[0][1:3]
+        if None in self.input_size:
+            raise Exception(f'model {args[MODEL_NAME]} has a dynamic input size, which is not supported by the MCT')
+        super().__init__(args)
+
+    def get_keras_apps_model(self, model_name: str):
+        """
+        Extracts the model package and model name from the input string: <package>.<model>
+        The model package class contains the 'preprocess_input' method for preprocessing input images
+        One exception is the MobileNetV3 models that are located in the parent model (tf.keras.applications)
+
+        Args:
+            model_name (str): A string containing the model to quantize in the following format: <package>.<model> (e.g. mobilenet_v2.MobileNetV2)
+
```

Comment:
About the<package>.<model>. it's not nice.. 
but I don't see other option.. Keras has 2 interfaces:
tf.keras.applications.MobileNetV2 and tf.keras.applications.mobilenet_v2.MobileNetV2, but indeed the preprocessing is found in tf.keras.applications.mobilenet_v2.preprocessing...
A suggestion, Isn't it possible to provide only the "package" and to get the model name out of there?